packages/http/src/util/runtimeExpression.ts

```diff
@@ -0,0 +1,128 @@
+import { IHttpRequest, IHttpResponse } from '../types';
+import * as Option from 'fp-ts/lib/Option';
+import { lookup } from 'fp-ts/lib/Array';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get as _get } from 'lodash';
+import { pointerToPath } from '@stoplight/json';
+
+export function resolveRuntimeExpressions(input: string, request: IHttpRequest, response: IHttpResponse) {
+  return input.replace(/{(.+?)}/g, (_, expr) =>
+    pipe(
+      resolveRuntimeExpression(expr, request, response),
+      Option.getOrElse(() => '')
+    )
+  );
+}
+
+export function resolveRuntimeExpression(
+  expr: string,
+  request: IHttpRequest,
+  response: IHttpResponse
+): Option.Option<string> {
+  const parts = expr.split(/[.#]/);
+
+  function isPart(idx: number, type: string) {
+    return pipe(
+      lookup(idx, parts),
+      Option.chain(part =>
+        pipe(
+          part,
+          Option.fromPredicate(part => part === type)
+        )
+      )
+    );
+  }
+
+  function tryMethod() {
+    return pipe(
+      isPart(0, '$method'),
+      Option.map(() => String(request.method))
+    );
+  }
+
+  function tryStatusCode() {
+    return pipe(
+      isPart(0, '$statusCode'),
+      Option.map(() => String(response.statusCode))
+    );
+  }
+
+  function tryRequestHeader() {
+    return pipe(
+      isPart(1, 'header'),
+      Option.chain(() => lookup(2, parts)),
+      Option.chain(part => Option.fromNullable(request.headers && request.headers[part]))
+    );
+  }
+
+  function tryRequestQuery() {
+    return pipe(
+      isPart(1, 'query'),
+      Option.chain(() => lookup(2, parts)),
+      Option.chain(part => Option.fromNullable(request.url.query && request.url.query[part]))
```

Comment:
Same as above, maybe it's better to use `Option.fromNullable` and `Option.mapNullable`

Reply:
Done 👍